src/actions/ast.js

```diff
@@ -51,13 +52,8 @@ export function setSourceMetaData(sourceId: SourceId) {
 
 export function setSymbols(sourceId: SourceId) {
   return async ({ dispatch, getState }: ThunkArgs) => {
-    const source = getSource(getState(), sourceId);
-    if (
-      !source ||
-      !source.text ||
-      source.isWasm ||
-      hasSymbols(getState(), source)
-    ) {
+    const source = getSourceFromId(getState(), sourceId);
```

Comment:
Minor thing, storing the state as a variable saves a little from having to get it twice here.